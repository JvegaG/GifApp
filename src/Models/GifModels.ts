

export interface ImagesProps {
    id: string;
    imageUrl: string;
    title: string;
}