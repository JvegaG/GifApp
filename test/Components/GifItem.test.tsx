import React from 'react'
import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/Components/_Component.index'
import { ImagesProps } from '../../src/Models/GifModels'

describe('<GifItem> Testing', () => {

    const imageTest: ImagesProps = {
        id: '2',
        title: 'Goku',
        imageUrl: 'https://dragonball.com/goku'
    }

    test('Create snapshot', () => {
        const { container } = render(<GifItem {...imageTest} />,);

        expect(container).toMatchSnapshot();
    })


    test('Analize if image contains fields correctly', () => {
        render(<GifItem {...imageTest} />,);

        // expect((screen.getByRole('img') as HTMLImageElement).src).toBe(imageTest.imageUrl);
        // expect((screen.getByRole('img') as HTMLImageElement).alt).toBe(imageTest.title);

        const { src, alt} = screen.getByRole('img') as HTMLImageElement;

        expect(src).toBe(imageTest.imageUrl);
        expect(alt).toBe(imageTest.title);

    })

    test('Analize if contains title', () => {
        render(<GifItem {...imageTest} />,);

        expect(screen.getByText(imageTest.title)).toBeTruthy();

    })


})