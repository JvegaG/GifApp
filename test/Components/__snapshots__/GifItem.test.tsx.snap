// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem> Testing Create snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://dragonball.com/goku"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
